php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -850,9 +850,9 @@ private function convertJsonValueToProtoValue(
         }
     }
 
-    private function mergeFromJsonArray($array)
+    private function mergeFromJsonObject($object)
```

Comment:
Is this rename intentional?

Reply:
The original mergeFromJsonArray is for implementation only. What's the benefit to use object instead of array?

Reply:
When the field is a map PHPs type system will convert a key with string `"1"` to the number `1` when decoding as an associative array, which will cause an exception with Protobuf's type checking. Decoding as an object preserving the key types.